webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala

```diff
@@ -362,6 +363,50 @@ class Boot extends Loggable {
     // the session itself.
     LiftRules.noCometSessionCmd.default.set(() => JsCmd.unitToJsCmd(()))
 
+    // Session expiration handler that ignores comet requests
+    val ForcedSessionTimeout = (sessions: Map[String, SessionInfo], delete: SessionInfo => Unit) => {
+      for ((id, info) <- sessions) {
+        info.session.httpSession.foreach(s => {
+          s.attribute("lastNonCometAccessedTime") match {
+            case lastNonCometAccessedTime: Long =>
+              val inactiveFor = millis - lastNonCometAccessedTime
+              LiftRules.sessionInactivityTimeout.vend.foreach(timeout => {
+                ApplicationLogger.warn(s"TIMEOUT: $id: $inactiveFor: $timeout ($info)")
+                if (inactiveFor > timeout) {
+                  // Ask for session termination
+                  // Race condition on destroySession that prevents destroying the underlying
+                  // HTTPSession?
+                  //info.session.destroySession()
+                  info.session.httpSession.foreach(s => s.terminate)
+                  // This only cleans up the session at lift level and unlink underlying session
+                  // but does nothing on it.
+                  delete(info)
+                }
+              })
+            case _ => ()
+          }
+        })
+      }
+    }
+    // Set session cleaner
+    SessionMaster.sessionCheckFuncs = SessionMaster.sessionCheckFuncs
+      ::: List(ForcedSessionTimeout)
+
+    // Store access time for user idle tracking
+    LiftRules.onBeginServicing.append((r: Req) => {
+      ApplicationLogger.warn(s"URL: ${r.uri}")
+      // This filters out lift-related XHR only, which is exactly what we want
+      if (r.standardRequest_?) {
+        ApplicationLogger.warn(s"URL: ${r.uri}: lastNonCometAccessedTime")
+        LiftRules.getLiftSession(r).httpSession
+          .foreach(s => s.setAttribute("lastNonCometAccessedTime", millis))
+      }
+    })
+
+    // 5 minutes
+    LiftRules.sessionInactivityTimeout.default.set(Full(1000 * 300L))
```

Comment:
This will need to be a parameter from rudder-web.properties

Reply:
this is still a wip, indeed